### PR TITLE
[Android] The text within buttons is slightly visible when buttons ar…

### LINKED ISSUE #19199: Text color in disable state buttons [Android]
# Design Feedback 2.27

## [Android] Disable state buttons have a wrong colored text
Different instances where the text color of a disabled button is wrong, making it hard to read:

- **Address for permissions screen:**

Screenshot 2024-03-12 alle 14 32 35

>
- **Join community:**

Screenshot 2024-03-12 alle 14 34 01

>
- **View profile:**

Screenshot 2024-03-12 alle 14 35 05

### Expected behavior

Screenshot 2024-03-12 alle 14 36 45

### Reproduction

1. Find a disabled button anywhere in the app from an Android device

### Additional Information

- Status version: nightly 
- Operating System: Android 8.1

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
…e in a disabled state

fixes #19199 

The previous implementation used opacity to render the disabled state, And opacity on Android as an attribute doesn't respect children's color and applies the opacity values to the entirety of the view.
While on iOS it only happens on the background.
This PR converts the hexadecimal value of the color and adds the opacity value for it, To have the same behavior for the disabled state between iOS and Android (Fading the background only and not the children)

### Summary
image
screenshot

status: ready

### CODE REVIEW

File: src/quo/components/buttons/button/style.cljs [hunk 1]
```diff
@@ -55,61 +56,64 @@
...
-          :border-color       border-color
-          :border-width       (when border-color 1)}
-         (when icon-only?
-           {:width size})
-         (when border-color
-           {:border-color border-color
-            :border-width 1})
-         (when disabled?
-           {:opacity 0.3})))
+  (let [disabled-and-rgb-hex? (and disabled? (colors/rgb-hex? background-color))]
```
Comment (ibrkhalil) [inline src/quo/components/buttons/button/style.cljs:59]: Added this `let` definition

File: src/quo/components/buttons/button/style.cljs [hunk 2]
```diff
@@ -55,61 +56,64 @@
...
+                                    12))
+            :border-color       border-color
+            :border-width       (when border-color 1)}
+           (when icon-only?
+             {:width size})
+           (when border-color
+             {:border-color border-color
+              :border-width 1})
+           (when-not disabled-and-rgb-hex?
+             {:opacity 0.3}))))
```
Comment (ibrkhalil) [inline src/quo/components/buttons/button/style.cljs:119]: Fallback to the older opacity approach if the button is disabled and the value of the background has an opacity.

File: src/quo/components/buttons/button/style.cljs [hunk 3]
```diff
@@ -55,61 +56,64 @@
...
+                                  (case size
+                                    56 0
+                                    40 9
+                                    32 5
+                                    24 0
+                                    9))
+            :overflow           :hidden
+            :background-color   (if disabled-and-rgb-hex?
+                                  (colors/hex->rgba background-color 0.3)
+                                  background-color)
```
Comment (ibrkhalil) [inline src/quo/components/buttons/button/style.cljs:102]: Relevant changes.

File: src/quo/components/buttons/button/style.cljs [hunk 4]
```diff
@@ -1,4 +1,5 @@
-(ns quo.components.buttons.button.style)
+(ns quo.components.buttons.button.style
```
Comment (ibrkhalil) [inline src/quo/components/buttons/button/style.cljs:1]: I commented on relevant changes on this file.

File: src/quo/foundations/colors.cljs [hunk 5]
```diff
@@ -299,6 +302,30 @@
...
+  ([s alpha-value]
+   (let [add-ending-parenthesis (fn [input-string]
+                                  (str input-string
+                                       (if alpha-value
+                                         (str ", " alpha-value ")")
+                                         ")")))]
+     (->> (subs s 1)
+          (partition-all 2)
+          (map #(apply (comp str (fn [number]
+                                   (js/parseInt number 16)) str) %))
```
Comment (ibrkhalil) [inline src/quo/foundations/colors.cljs:324]: JS conversion from hexadecimal to decimal, Tried using `native-module.core/hex-to-number` in this ns, But component-specs didn't like it. :)

File: src/quo/components/buttons/button/style.cljs [hunk 6]
```diff
@@ -55,61 +56,64 @@
...
+                                    (if border-color 8 9)))
+            :padding-bottom     (when-not (or icon-only? icon-left icon-right)
+                                  (case size
+                                    56 0
+                                    40 9
+                                    32 5
+                                    24 0
+                                    9))
+            :overflow           :hidden
+            :background-color   (if disabled-and-rgb-hex?
```
Comment (J-Son89) [inline src/quo/components/buttons/button/style.cljs]: why do we need to check if it is `hex` as well? - can't we just check if it is disabled and app the opacity of 0.3 if so?

Comment (J-Son89) [reply src/quo/components/buttons/button/style.cljs]:
```
:background-color (if disabled? (colors/alpha background-color 0.3)  background-color)
```

Comment (ibrkhalil) [reply src/quo/components/buttons/button/style.cljs]: Thank you so much Jamie, I totally forgot the we had an alpha function in the colors ns :)
Removing the conversion I created :)

Comment (ibrkhalil) [reply src/quo/components/buttons/button/style.cljs]: The check was to make sure the hex color is only of 7 digits (Hex color without opacity value.)

**Review by J-Son89 — APPROVED** [review]

**Review by Parveshdhull — APPROVED** [review]

**Pull request merged**